File: Week 2/Problem Set/Readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//index = 0.0588 * L - 0.296 * S - 15.8
int letter_counter(string text);
int word_counter(string text);
int sentence_counter(string text);

int main(void)
{
    string text = get_string("Text: ");
    // One variable to each function
    int letters = letter_counter(text);
    int words = word_counter(text);
    int sentences = sentence_counter(text);
    // Use casting to make the values letters, words, sentences into float number, so we don't lose any valu in the division
    float index = 0.0588 * ((float)letters / (float)words * 100) - 0.296 * ((float)sentences / (float)words * 100) - 15.8;
    // The round will round the floating point number to it's nearest integet
    if (round(index) < 1)
    {
        printf("Before Grade 1\n");
    }
    if (round(index) > 16)
    {
        printf("Grade 16+\n");
    }
    if (round(index) > 1 && round(index) < 16)
    {
        printf("Grade %.0f\n", round(index));
    }
}

int letter_counter(string text)
{
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            // Counts every time of word using the isalpha function
            letters++;
        }
    }
    return letters;
}
int word_counter(string text)
{
    int words = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            // Counts one word for every space
            words++;
        }
    }
    return words + 1;
    // Returns words + 1 because this loop won't count the last word, the last word wont have a space like the others
}
int sentence_counter(string text)
{
    int sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            // Compare the letter if this three punctuations
            sentences++;
        }
    }
    return sentences;
}

File: Week 6/2. Problem Set/5. Readability/readability.py
from cs50 import get_string


def main():

    text = get_text()
    letters = get_letters(text)
    words = get_words(text)
    sentences = get_sentences(text)
    # We already been gave the formula
    # Since python will store the values as int
    # We have to cast they to floats
    index = 0.0588 * (float(letters) / float(words) * 100) - 0.296 * (float(sentences) / float(words) * 100) - 15.8
    index = round(index)

    if index < 1:
        print("Before Grade 1")
    elif index > 1 and index < 16:
        print("Grade " + str(index))
    elif index > 16:
        print("Grade 16+")


def get_text():

    text = get_string("Text: ")

    return text


def get_letters(text):

    letters = 0
    for i in range(len(text)):
        # Uses the function isalpha to count the words
        if text[i].isalpha():
            letters += 1

    return letters


def get_words(text):

    spaces = 0
    for i in range(len(text)):
        if text[i] == ' ':
            spaces += 1
    # Do direct comparisons to spaces
    # We have to add a +1 to the value because the loops
    # Fail to count the last word
    return spaces + 1


def get_sentences(text):

    sentences = 0
    for i in range(len(text)):
        # Do direct comparissons to . ! and ?
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentences += 1

    return sentences


if __name__ == "__main__":
    main()